src/ghga_connector/config.py

```diff
@@ -115,47 +115,52 @@ async def set_runtime_config(client: httpx.AsyncClient):
     - wps_api_url
     - dcs_api_url
     - ucs_api_url
+
+    Raises:
+        WellKnownValueNotFound: If one of the well-known values is not found in the
+            response from the WKVS.
+        BadResponseCodeError: If the status code returned is not 200
+        ConnectionFailedError: If the request fails due to a timeout/connection problem
+        RequestFailedError: If the request fails for any other reason
     """
-    ghga_pubkey = await _get_wkvs_value(client, value_name="crypt4gh_public_key")
-    wps_api_url = (await _get_wkvs_value(client, value_name="wps_api_url")).rstrip("/")
-    dcs_api_url = (await _get_wkvs_value(client, value_name="dcs_api_url")).rstrip("/")
-    ucs_api_url = (await _get_wkvs_value(client, value_name="ucs_api_url")).rstrip("/")
+    values = await _get_wkvs_values(client)
+    for value_name in [
+        "crypt4gh_public_key",
+        "wps_api_url",
+        "dcs_api_url",
+        "ucs_api_url",
+    ]:
+        if value_name not in values:
+            raise exceptions.WellKnownValueNotFound(value_name=value_name)
 
     async with (
-        set_context_var(ghga_pubkey_var, ghga_pubkey),
-        set_context_var(wps_api_url_var, wps_api_url),
-        set_context_var(dcs_api_url_var, dcs_api_url),
-        set_context_var(ucs_api_url_var, ucs_api_url),
+        set_context_var(ghga_pubkey_var, values["crypt4gh_public_key"]),
+        set_context_var(wps_api_url_var, values["wps_api_url"].rstrip("/")),
+        set_context_var(dcs_api_url_var, values["dcs_api_url"].rstrip("/")),
+        set_context_var(ucs_api_url_var, values["ucs_api_url"].rstrip("/")),
     ):
         yield
 
 
-async def _get_wkvs_value(client: httpx.AsyncClient, *, value_name: str) -> Any:
-    """Retrieve a value from the well-known-value-service.
-
-    Args:
-        value_name (str): the name of the value to be retrieved
+async def _get_wkvs_values(client: httpx.AsyncClient) -> dict[str, Any]:
+    """Retrieve a value from the well-known-value-service using the supplied client.
 
     Raises:
-        WellKnownValueNotFound: when a 404 response is received from the WKVS
-        KeyError: when a successful response is received but doesn't contain the expected value
+        BadResponseCodeError: If the status code returned is not 200
+        ConnectionFailedError: If the request fails due to a timeout/connection problem
+        RequestFailedError: If the request fails for any other reason
     """
-    url = f"{CONFIG.wkvs_api_url}/values/{value_name}"
+    url = f"{CONFIG.wkvs_api_url}/values/"
```

Comment:
There's no trailing slash in the WKVS route, so for consistency it could also be removed here.